reconcilers/stash_test.go

```diff
@@ -84,10 +84,87 @@ func TestStash_RetrieveValue_UndecoratedContext(t *testing.T) {
 }
 
 func TestStash_RetrieveValue_Undefined(t *testing.T) {
-	ctx := WithStash(context.TODO())
+	ctx := WithStash(context.Background())
 	var key StashKey = "stash-key"
 
 	if value := RetrieveValue(ctx, key); value != nil {
 		t.Error("expected RetrieveValue() to return nil for undefined key")
 	}
 }
+
+func TestStasher(t *testing.T) {
+	ctx := WithStash(context.Background())
+	stasher := NewStasher[string]("my-key")
```

Comment:
**[nit]:** I think testing against a stashed value of type `string` is sufficient. Then again maybe the test becomes more interesting when using a non-primitive type like `Example`. In particular I am thinking about `RetrieveOrEmpty`.

Reply:
All types in go have a default empty value. There's an example using a vanilla struct here:

https://github.com/reconcilerio/runtime/pull/546/files#diff-21cd712a5bbfcb003ce654296346a4008831cce30760e9b6d2d1fb61a0bb5f4dR235

Reply:
I get it. ty